### ISSUE TITLE
allow a (remote) Git commit as installation source

### ISSUE THREAD

**Issue by amigalemming** [haskell/cabal#6067]
I'd like to check whether an issue has been fixed in a current version of a Cabal package.
Thus I'd like to compile the package in a certain state, given by a certain Git commit, like so:

~~~~
$ cabal install https://github.com/pcapriotti/optparse-applicative/commit/1a38782b12373eaae4348584e6a6ff625f9afa89
~~~~

**Comment by 23Skidoo** [comment]
Right now you can do the following:

```
$ wget https://github.com/pcapriotti/optparse-applicative/archive/1a38782b12373eaae4348584e6a6ff625f9afa89.tar.gz
$ tar xvf 1a38782b12373eaae4348584e6a6ff625f9afa89.tar.gz
$ cd optparse-applicative-1a38782b12373eaae4348584e6a6ff625f9afa89
$ cabal build
```

**Comment by amigalemming** [comment]
So I could also do:

~~~~
$ cabal install https://github.com/pcapriotti/optparse-applicative/archive/1a38782b12373eaae4348584e6a6ff625f9afa89.tar.gz
~~~~

However, Cabal does not like the directory name (without version) in the archive.
I remember this was also a problem with `darcs dist`.

**Comment by 23Skidoo** [comment]
Yeah, should be possible to tweak the logic to go one level down in the archive if it contains only a single top-level directory.

**Comment by amigalemming** [comment]
#1822

**Comment by 23Skidoo** [comment]
Closing as a duplicate then.

**Comment by 23Skidoo** [comment]
Closed in favour of #1822.

**Comment by hvr** [comment]
@23Skidoo actually I think #5522 would be more in the spirit of the "install from git" suggestion; note the "stretch goal" in its issue description
